src/runtime_src/core/tools/common/TestRunner.cpp

```diff
@@ -85,6 +99,33 @@ TestRunner::TestRunner (const std::string & test_name,
   //Empty
 }
 
+boost::property_tree::ptree 
+TestRunner::startTest(std::shared_ptr<xrt_core::device> dev)
+{
+  XBUtilities::BusyBar busy_bar("Running Test", std::cout);
+  busy_bar.start(XBUtilities::is_escape_codes_disabled());
+  bool is_thread_running = true;
+
+  boost::property_tree::ptree result;
+
+  // Start the test process
+  std::thread test_thread([&] { runTestInternal(dev, result, this, is_thread_running); });
+  // Wait for the test process to finish
+  while (is_thread_running) {
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    try {
+      busy_bar.check_timeout(MAX_TEST_DURATION);
+    } catch (const std::exception&) {
+      test_thread.detach();
+      throw;
```

Comment:
What is this throw for?

Reply:
I'll catch this in SubCmdValidate and fail the test if there's an exception like this (e.g. there's an issue with checking timeout).